src/main/java/com/faforever/moderatorclient/ui/main_window/LadderMapPoolController.java

```diff
@@ -176,40 +235,50 @@ private void bindListViewSelectionToSelectedMapProperty(ListView<MapVersionFX> l
+    private void bindSelectedMapPropertyToAddRemoveButtons(ObservableList<MapPoolAssignmentFX> mapList, AddBracketController controller, MapPoolFX bracketPool) {
         selectedMap.addListener((observable, oldValue, newValue) -> {
-            if (newValue == null) return;
+            if (newValue == null) {
+                return;
+            }
             Button addButton = controller.addToBracketButton;
             Button removeButton = controller.removeFromBracketButton;
             // enable add/remove buttons
-            addButton.setDisable(mapList.contains(newValue));
-            removeButton.setDisable(!mapList.contains(newValue));
+            addButton.setDisable(mapList.stream().anyMatch(assignmentFX -> matchingPoolAssignmentPredicate.test(newValue, assignmentFX)));
+            removeButton.setDisable(!addButton.isDisable());
             //bind actions for add and remove buttons
             addButton.setOnAction(event -> {
-                    mapList.add(newValue);
-                    addButton.setDisable(true);
-                    removeButton.setDisable(false);
+                MapPoolAssignmentFX newAssignment = new MapPoolAssignmentFX();
+                newAssignment.setWeight(1);
+                newAssignment.setMapParams(newValue.getMapParams());
+                newAssignment.setMapVersion(newValue.getMapVersion());
+                newAssignment.setMapPool(bracketPool);
+                mapList.add(newAssignment);
+                addButton.setDisable(true);
+                removeButton.setDisable(false);
             });
             removeButton.setOnAction(event -> {
-                    mapList.remove(newValue);
-                    // prevents the client from changing the selection when removing the currently selected map
-                    selectedMap.setValue(newValue);
-                    removeButton.setDisable(true);
-                    addButton.setDisable(false);
+                mapList.removeIf(assignmentFX -> matchingPoolAssignmentPredicate.test(newValue, assignmentFX));
+                // prevents the client from changing the selection when removing the currently selected map
+                selectedMap.setValue(newValue);
+                removeButton.setDisable(true);
+                addButton.setDisable(false);
             });
         });
     }
 
     private String getBracketRatingString(MatchmakerQueueMapPoolFX bracket) {
         var min = bracket.getMinRating();
         var max = bracket.getMaxRating();
-        if (min == 0) return String.format("<%d", (int)max);
-        if (max == 0) return String.format(">%d", (int)min);
-        return String.format("%d - %d", (int)min, (int)max);
+        if (min == 0) return String.format("<%d", (int) max);
+        if (max == 0) return String.format(">%d", (int) min);
+        return String.format("%d - %d", (int) min, (int) max);
```

Comment:
Why don't you cast it to int on variable assignment? Also we don't use var in this project.
I like var in general, but would only use it consistently for new projects.

Reply:
ah that var was already there and I didnt notice